eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,181 @@
+                if (ast.getNextSibling() == null) {
+                    break;
+                }
+                // Checks no use any external logger class.
+                if (invalidLogger != null) {
+                    log(ast, String.format(NOT_CLIENT_LOGGER_ERROR, CLIENT_LOGGER_PATH, invalidLogger));
+                }
+
+                className = ast.getNextSibling().getText();
+                break;
+            case TokenTypes.LITERAL_NEW:
+                checkLoggerInstantiation(ast);
+                break;
+            case TokenTypes.VARIABLE_DEF:
+                checkLoggerNameMatch(ast);
+                break;
+            case TokenTypes.METHOD_CALL:
+                final DetailAST dotToken = ast.findFirstToken(TokenTypes.DOT);
+                if (dotToken == null) {
+                    return;
+                }
+                final String methodCallName = FullIdent.createFullIdentBelow(dotToken).getText();
+                if (methodCallName.startsWith("System.out") || methodCallName.startsWith("System.err")) {
+                    log(ast, String.format("Do not use Java System class for logging. Use ClientLogger in ''%s'' instead.", CLIENT_LOGGER_PATH));
+                }
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * check if the VARIABLE_DEF AST node type is 'ClientLogger'.
+     * @param varDefAST VARIABLE_DEF AST node
+     * @return true if the variable type is 'ClientLogger'.
+     */
+    private boolean isTypeClientLogger(DetailAST varDefAST) {
+        DetailAST typeAST = varDefAST.findFirstToken(TokenTypes.TYPE);
+        if (typeAST == null) {
+            return false;
+        }
+        return TokenUtil.findFirstTokenByPredicate(typeAST, node ->
+           node.getType() == TokenTypes.IDENT && node.getText().equals(CLIENT_LOGGER)
+        ).isPresent();
+    }
+
+    /**
+     * Check if instantiating a matched class name for the same class.
+     *
+     * @param literalNewToken LITERAL_NEW node
+     */
+    private void checkLoggerInstantiation(DetailAST literalNewToken) {
+        final DetailAST identToken = literalNewToken.findFirstToken(TokenTypes.IDENT);
+        // Not named 'logger'
+        if (identToken == null || !identToken.getText().equals(CLIENT_LOGGER)) {
+            return;
+        }
+        // Edge cases
+        final DetailAST elistToken = literalNewToken.findFirstToken(TokenTypes.ELIST);
+        final DetailAST exprToken = elistToken.findFirstToken(TokenTypes.EXPR);
+        if (exprToken.getFirstChild().getType() != TokenTypes.DOT) {
+            return;
+        }
+        // Check instantiation of ClientLogger
+        String containerClassName = FullIdent.createFullIdentBelow(exprToken).getText();
+        containerClassName = containerClassName.substring(0, containerClassName.length() - 6);
```

Comment:
Add a comment to clarify why you are using `containerClassName.length() - 6` as the end index.